torch/distributed/pipelining/schedules.py

```diff
@@ -612,6 +649,88 @@ def __init__(
             if self._has_backward:
                 stage._prepare_backward_infra(n_microbatches)
 
+    def _dump_csv(self, filename):
+        """Dump a CSV representation of the schedule into a file with the provided filename.
+        This API will most likely get renamed/refactored so is marked as internal for now.
+        """
+        with open(filename, "w", newline="") as csvfile:
+            writer = csv.writer(csvfile)
+            for rank in self.pipeline_order:
+                writer.writerow(self.pipeline_order[rank])
+
+    def _validate_schedule(self):
+        # TODO(whc) this should be merged with the logic in test_schedule.py#L453-L554
```

Comment:
nit: the `_validate_schedule` method could seem a bit heavy to be in a base class -- is there a chance to move it to a separate utility and be in functional form, e.g. `validate(schedule)`?

Another consideration is whether every schedule should support this method -- are we now rewriting every schedule to have a field like `pipeline_order`?

Reply:
I intended to put this in the schedule multi class, since currently that is the one that implement step via pipeline order. If I put it in the base class instead we can move it.

I actually did think about eventually moving more schedules to this format but for now it's evolving.